Assets/MixedRealityToolkit/_Core/Sharing/AnchorControl/AnchorSynchronizer.cs

```diff
@@ -0,0 +1,236 @@
+﻿using Pixie.AppSystems;
+using Pixie.Core;
+using Pixie.DeviceControl;
+using Pixie.StateControl;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Pixie.AnchorControl
+{
+    public class AnchorSynchronizer : MonoBehaviourSharingApp, ISharingAppObject, IAnchorSynchronizerClient, IAnchorSynchronizerServer
+    {
+        public const bool ApplyAlignmentRotation = true;
+        public const bool ApplyAlignmentScale = false;
+        public const bool RotateOnYAxisOnly = true;
+
+        public AnchorSyncStateEnum State { get { return state; } }
+        
+        [SerializeField]
+        private AnchorSyncStateEnum state = AnchorSyncStateEnum.Stopped;
+        private Dictionary<short, Transform> sharedAnchorTransformLookup = new Dictionary<short, Transform>();
+        private Dictionary<short, Transform> userAnchorTransformLookup = new Dictionary<short, Transform>();
+        private List<Transform> sharedAnchorTransformList = new List<Transform>();
+        private List<Transform> userAnchorTransformList = new List<Transform>();
+        private Transform sceneRootTransform;
+
+        public void CreateAnchorStates(IAnchorDefinitions definitions, IAppStateReadWrite appState)
+        {
+            switch (state)
+            {
+                case AnchorSyncStateEnum.Stopped:
+                    break;
+
+                default:
+                    throw new Exception("Can't do this when state is " + state);
+            }
+
+            state = AnchorSyncStateEnum.CreatingSharedAnchors;
+            StartCoroutine(CreateAnchorStatesInternal(definitions, appState));
+        }
+
+        private IEnumerator CreateAnchorStatesInternal(IAnchorDefinitions definitions, IAppStateReadWrite appState)
```

Comment:
Do we need to dynamically monitor changes for AppState UserSlots? This function looks like it is only intended to be called during initialization, but i feel like we may want to create/destroy user anchors as users leave and enter the stage/session?